### PR TITLE
[ROCm] Adding ROCm support for the softmax op

### PR DESCRIPTION
**Pull request by deven-amd**
This PR adds ROCm support for the softmax op

The changes in this PR are trivial, please review and merge...thanks.

----------------------------------------------------------

@tatianashp @whchung @chsigg

### CODE REVIEW

**Review by whchung — DISMISSED** [review]

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc [hunk 1]
```diff
@@ -142,7 +148,7 @@ class SoftmaxOpGPU : public OpKernel {
     OP_REQUIRES_OK(context, context->forward_input_or_allocate_output(
                                 {0}, 0, logits_in_.shape(), &softmax_out));
 
-    const cudaStream_t& cu_stream = GetCudaStream(context);
+    const gpuStream_t& cu_stream = GetGpuStream(context);
```
Comment (chsigg) [inline tensorflow/core/kernels/softmax_op_gpu.cu.cc]: auto?

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc [hunk 2]
```diff
@@ -174,12 +180,12 @@ class SoftmaxOpGPU : public OpKernel {
...
 
-      DoRowReduction<acc_type, cub::Sum, InputIterType>(
+      DoRowReduction<acc_type, gpuprim::Sum, InputIterType>(
           context, const_cast<acc_type*>(sum_probs.flat<acc_type>().data()),
           input_itr, rows, cols);
 
-      TF_CHECK_OK(CudaLaunchKernel(
-          GenerateNormalizedProb<T, acc_type>, numBlocks, numThreads, 0,
+      TF_CHECK_OK(GpuLaunchKernel(
+          (GenerateNormalizedProb<T, acc_type>), numBlocks, numThreads, 0,
```
Comment (chsigg) [inline tensorflow/core/kernels/softmax_op_gpu.cu.cc]: Remove extra parenthesis.

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc [hunk 3]
```diff
@@ -155,17 +161,17 @@ class SoftmaxOpGPU : public OpKernel {
...
 
       const int numThreads = 128;
       const int numBlocks = Eigen::divup(rows * cols, numThreads);
 
-      cub::CountingInputIterator<int> counting_iterator(0);
-      typedef cub::TransformInputIterator<acc_type,
-                                          SubtractAndExpFunctor<T, acc_type>,
-                                          cub::CountingInputIterator<int>>
+      gpuprim::CountingInputIterator<int> counting_iterator(0);
+      typedef gpuprim::TransformInputIterator<
```
Comment (chsigg) [inline tensorflow/core/kernels/softmax_op_gpu.cu.cc]: You are inheriting this, but we prefer 'using' these days.

**Review by chsigg — DISMISSED** [review]

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc (hunk 1 above)
Comment (deven-amd) [reply tensorflow/core/kernels/softmax_op_gpu.cu.cc]: done

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc (hunk 2 above)
Comment (deven-amd) [reply tensorflow/core/kernels/softmax_op_gpu.cu.cc]: done

File: tensorflow/core/kernels/softmax_op_gpu.cu.cc (hunk 3 above)
Comment (deven-amd) [reply tensorflow/core/kernels/softmax_op_gpu.cu.cc]: done

**Review by whchung — APPROVED** [review]

**Review by chsigg — APPROVED** [review]
